### PR TITLE
[Logger] Add redact step

### PR DESCRIPTION
**Pull request by dayo09**
This commit adds redact step to a Logger.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

---

For #1609

### CODE REVIEW

**Review by jyoungyun — DISMISSED** [review]
LGTM

**Comment by dayo09** [comment]
@Samsung/one-vscode_committers PTAL :-D

File: src/Utils/Logger.ts [hunk 1]
```diff
@@ -33,26 +33,36 @@ const isDebugMode = process.env.VSCODE_DEBUG_MODE === "true";
...
     }
-  }
-  const msg = logStrList.join(" ");
+    return logStrList.join(" ");
+  };
+
+  const redact = (msg: string) => {
+    // Replace github tokens with ********
+    const prefix = "ghp_";
+    const regex = new RegExp(`${prefix}[a-zA-Z0-9]+`, "g");
```
Comment (batcheu) [inline src/Utils/Logger.ts:61]: How about to add the regular expression for ['Fine Grained Personal Acess Token'](https://docs.github.com/ko/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens) either?
I'm not sure whether RegExp() supports 'ORed' expression, if so we can add it as OR-ed option as like,
```
const prefix_classic = "ghp_";
const prefix_fgp = "github_pat_";
`^(${prefix_classic}[a-zA-Z0-9]+|${prefix_fgp}_[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59})`
```

Comment (dayo09) [reply src/Utils/Logger.ts:61]: @batcheu Look like a good idea to add the redact for fine grained token :-D

BTW I didn't find the string length of classic token so I supposed to redact all the strings starting with 'ghp_'. 
And I cannot find the string length of fine grained token written somewhere in github page either. 

If you have found some, could you let me know the link?

If not, let me simply add find-grained Regex as github_pat_[a-zA-Z0-9]+.

Comment (dayo09) [reply src/Utils/Logger.ts]: @batcheu Applied :-D

Comment (batcheu) [reply src/Utils/Logger.ts]: Oops, I didn't notice that you've added comments. 😮
Sorry for late reply 😢 

I just check the regular expression of new type of access token in below link.
* [Github token validation regular expression](https://gist.github.com/magnetikonline/073afe7909ffdd6f10ef06a00bc3bc88)

And I also found below GITHUB's official annoucement related with access token today.
* https://github.blog/changelog/2021-03-31-authentication-token-format-updates-are-generally-available/

> The length of our tokens is remaining the same for now. 
> However, GitHub tokens will likely increase in length in future updates, so integrators should plan to support tokens up to 255 characters after June 1, 2021.

So, I think that your suggestion is more proper than fixed length expression.

**Review by batcheu — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by dayo09** [comment]
@Samsung/one-vscode_committers PTAL :-D

**Pull request merged**